### ISSUE TITLE
Layout of *.pl files in tarball and installation

### ISSUE THREAD

**Issue by jkeenan** [abeltje/Test-Smoke#32]
This is a question as much as it is an "issue".  The answer to the question should probably go in the FAQ (unless I've overlooked existing documentation somewhere.)

When I download a tarball of Test-Smoke from CPAN and unpack it, I find that in the top-level directory I have 18 ```.pl``` files:
```
[Test-Smoke-1.70] $ find . -maxdepth 1 -type f -name '*.pl'
./archiverpt.pl
./chkbcfg.pl
./configsmoke.pl
./mailrpt.pl
./patchtree.pl
./runsmoke.pl
./sendrpt.pl
./smokeperl.pl
./smokestatus.pl
./synctree.pl
./sysinfo.pl
./tsarchive.pl
./tsreport.pl
./tsrunsmoke.pl
./tssendrpt.pl
./tssmokeperl.pl
./tssynctree.pl
./W32Configure.pl
[Test-Smoke-1.70] $ find . -maxdepth 1 -type f -name '*.pl' | wc -l
      18
```
Each of these perl programs starts with a shebang line calling the system ```perl```.
```perl
#! /usr/bin/perl -w
```
Now let's assume that I install Test-Smoke with ```SMOKE_INSTDIR=~/p5smoke/install```.  It is my observation that the 18 Perl programs listed above get moved to ```install/Test/``` and that **all but one** of those programs gets installed in slightly modified form in ```install/```.
```
[install] $ pwd
/home/jkeenan/p5smoke/install
[install] $ ls *.pl |cat
W32Configure.pl
archiverpt.pl
chkbcfg.pl
configsmoke.pl
mailrpt.pl
patchtree.pl
runsmoke.pl
sendrpt.pl
smokeperl.pl
smokestatus.pl
synctree.pl
sysinfo.pl
tsarchive.pl
tsrunsmoke.pl
tssendrpt.pl
tssmokeperl.pl
tssynctree.pl

[install] $ ls *.pl | wc -l
      17
```
The files in ```install/``` differ from those now found in ```install/Test``` like this:
```diff
[install] $ diff -w Test/tsrunsmoke.pl ./tsrunsmoke.pl 
1c1,4
< #! /usr/bin/perl -w
---
> #!/usr/local/bin/perl -w
> 
> eval 'exec /usr/local/bin/perl -w -S $0 ${1+"$@"}'
>     if 0; # not running under some shell
```
Are my observations correct so far?

If they are correct, can you explain the rationale for this "double" installation?

And can you explain why ```tsreport.pl``` is not subject to this double installation?  Does ```Test/tsreport.pl``` (still) serve some special purpose?

Thank you very much.
Jim Keenan

**Comment by abeltje** [comment]
Hi Jim,

I can explain the ```tsreport.pl``` thing: it wasn't in the scripts list. That brings me to the changed "header" of files marked as script. This is what make-maker does for files marked as script. Instead of using ```/usr/bin/perl``` you used ```/usr/local/bin/perl``` to install the module.

So much for the easy part of this answer. The hard part is the question "Why are the scripts also copied to the ```install/Test``` directory?". Or in fact what makes MakeMaker put them in the ```pm_to_blib``` target of the ```Makefile```.

As we are close to "no-dot-in-at-inc", I'll have to reconsider the whole directory-layout of ```Test::Smoke```

I'll be working on that for the rest of this PerlTS :wink:

**Comment by jkeenan** [comment]
On 05/14/2017 06:59 AM, Abe Timmerman wrote:
> Hi Jim,
>
> I can explain the |tsreport.pl| thing: it wasn't in the scripts list.
> That brings me to the changed "header" of files marked as script. This
> is what make-maker does for files marked as script. Instead of using
> |/usr/bin/perl| you used |/usr/local/bin/perl| to install the module.
>
> So much for the easy part of this answer. The hard part is the question
> "Why are the scripts also copied to the |install/Test| directory?". Or
> in fact what makes MakeMaker put them in the |pm_to_blib| target of the
> |Makefile|.
>
> As we are close to "no-dot-in-at-inc", I'll have to reconsider the whole
> directory-layout of |Test::Smoke|
>

Thanks for responding.  The whole layout of files in install/ is 
difficult to understand.  I've actually begun to write a separate issue 
about that.

jimk

**Comment by abeltje** [comment]
Please don't ;) I'm working on a better structure. Please take a look at the ```no_dot_in_at_inc``` branch for my progress...

**Comment by abeltje** [comment]
That was released in Test-Smoke-1.71 (2017-05-14)
I'm closing this comment.
